src/addons/draggable/RowDragLayer.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component, PropTypes } from 'react';
+import { DragLayer } from 'react-dnd';
+import Selectors from '../data/Selectors';
+
+const layerStyles = {
+  cursor: '-webkit-grabbing',
+  position: 'fixed',
+  pointerEvents: 'none',
+  zIndex: 100,
+  left: 0,
+  top: 0,
+  width: '100%',
+  height: '100%'
+};
+
+function getItemStyles(props) {
+  const { currentOffset } = props;
+  if (!currentOffset) {
+    return {
+      display: 'none'
+    };
+  }
+
+  const { x, y } = currentOffset;
+  const transform = `translate(${x}px, ${y}px)`;
+  return {
+    transform: transform,
+    WebkitTransform: transform
+  };
+}
+
+class CustomDragLayer extends Component {
+
+  isDraggedRowSelected(selectedRows) {
+    let {item, rowSelection} = this.props;
+    if (selectedRows && selectedRows.length > 0) {
+      let key = rowSelection.selectBy.keys.rowKey;
+      return selectedRows.filter(r => r[key] === item.data[key]).length > 0;
+    }
+    return false;
+  }
+
+  getRows() {
+    let rows = [];
+    let {rowsCount, rowGetter} = this.props;
+    for (let j = 0; j < rowsCount; j++) {
+      rows.push(rowGetter(j));
+    }
+    return rows;
+  }
+
+  getDraggedRows() {
+    let draggedRows;
+    let {rowSelection} = this.props;
+    if (rowSelection && rowSelection.selectBy.keys) {
+      let rows = this.getRows();
```

Comment:
Will he get the rows every time we render?
Can the `rowCount` be big?
Can it damage the performance? If not looks ok to me, if it can do (quick profile to figure out?) I guess we should put the rows in state and update only when needed.